File: 2/password2.py
instuff = open("password.in", "r")
instuff1 = instuff.readlines()
instuff2 = []
for i in instuff1:
    stuff = []
    for s in i.split(" "):
        stuff.append(s)
    instuff2.append(stuff)

result = 0
for i in instuff2:
    start = int(i[0].split("-")[0])
    end = int(i[0].split("-")[1])
    letter = i[1].split(":")[0]
    if (i[2][start - 1] == letter) ^ (i[2][end - 1] == letter):
        result += 1

print(result)

File: 2/password2.cpp
#include <bits/stdc++.h>

int main()
{
    std::ifstream infile("password.in");
    std::string data;
    int result = 0;

    while (infile >> data)
    {
        auto p = data.begin();
        std::string start = "";
        std::string end = "";

        while (*p != '-')
        {
            start += *p;
            p++;
        }

        p++;

        while (p < data.end())
        {
            end += *p;
            p++;
        }

        char garbo, letter;
        infile >> letter;
        infile >> garbo;

        std::string pw = "";

        infile >> pw;

        p = pw.begin();
        int count = 0;

        if (p[std::stoi(start) - 1] == letter ^ p[std::stoi(end) - 1] == letter)
        {
            result++;
        }
    }

    std::cout << result;
}